File: editor/en/[2870]Minimum Number of Operations to Make Array Empty.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int minOperations(int[] nums) {
        Arrays.sort(nums);
        int count = 0;
        int left = 0;

        while (left < nums.length) {
            int right = left;

            while (right < nums.length && nums[left] == nums[right]) {
                right++;
            }

            int num = right - left;

            if (num == 1) {
                return -1;
            }

            count += (int) Math.ceil(num / 3.0);
            left = right;
        }

        return count;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
/*
class Solution {
    public int minOperations(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int count = 0;

        for (int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        for (int num : map.values()) {
            if (num == 1) {
                return -1;
            }

            count += (int) Math.ceil(num / 3.0);
        }

        return count;
    }
}
 */

File: editor/en/[2870]Minimum Number of Operations to Make Array Empty.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @return {number}
 */
var minOperations = function (nums) {
    let map = new Map();
    let count = 0;

    for (let num of nums) {
        map.set(num, (map.get(num) || 0) + 1);
    }

    for (let num of map.values()) {
        if (num === 1) {
            return -1;
        }

        if (num % 3 === 0) {
            count += num / 3;
        } else {
            count += Math.floor(num / 3) + 1;
        }
    }

    return count;
};
//leetcode submit region end(Prohibit modification and deletion)
